### DISCUSSION TITLE
i wanted to have a middleware, to check if there is a records with similar name before creating new record.

### DISCUSSION THREAD

**Discussion by getch04** [pocketbase/pocketbase#3198] [Q&A] [answered]
I wanted to have a middleware, to check if there is a record with a similar name before creating a new record.
   1. If the record is found with a similar name, it should update it.
   2. If there is no record with a similar name, it should create one as a new

**Comment by ganigeorgiev** [comment]
There is no "upsert" builtin web api.

You can try to send individual requests (one to check for existence and another one for create/update) OR create a custom route (see https://pocketbase.io/docs/use-as-framework/).

If you are using the SDKs, to retrieve a record by a non-id field value, you can use `getFirstListItem`. Here is an example with the JS SDK:

```js
let name = "..."
let data = { ... }

const record = await pb.collection("example").getFirstListItem(`name="${name}"`)

if (!record) {
    await pb.collection("example").create(data)
} else {
    await pb.collection("example").update(record.id, data)
}
```

Additionally, to ensure that only all records have a unique field value (in your case name), you can define a unique index for the specific field(s) from the _Admin UI > {collection} > Edit collection_ panel.

**Comment by getch04** [comment]
Ah, i see, One additional question Is there any way to accept a list of records(during creating), rather than one record at a time?

**Reply by ganigeorgiev** [reply, marked as answer]
No, bulk and transactional operations are currently not supported with the web APIs. For that you'll have to [register your own custom route](https://pocketbase.io/docs/js-routing/#registering-new-routes) and [manually persists the record(s)](https://pocketbase.io/docs/js-records).

**Reply by getch04** [reply]
thanks a lot
